workflow/util/unstructured.go

```diff
@@ -0,0 +1,704 @@
+/*
+NOTE: this file is a hack to make unstructured runtime objects work with inlined ParallelSteps
+It is taken from https://github.com/kubernetes/apimachinery/blob/v0.17.8/pkg/runtime/converter.go
+and should be removed once k8s.io/apimachinery can handle the changes made by
+https://github.com/argoproj/argo/pull/4828
```

Comment:
it would be easier to revert https://github.com/argoproj/argo/pull/4828 yes?

Reply:
Probably - I just didn't know if you guys wanted to do that

Reply:
this is a trade-off, and I think reverting it would be safer, even if it is a shame

Reply:
Makes sense. I think it's a shame as well

Reply:
Tagging @taharah as a heads up. The inlining is preventing an upgrade to apimachinery. Going to revert your commit to safely accommodate apimachinery v0.19.6 which kicks a panic when `ParallelSteps` are inlined 😞

Reply:
Thanks for tagging me. I'll checkout your changes and see if I can get it working again.

Reply:
Thanks @taharah!